packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx

```diff
@@ -210,6 +210,10 @@ export class DetailsList extends BaseComponent<IDetailsListProps, IDetailsListSt
 
     if (shouldResetSelection) {
       this._initialFocusedIndex = newProps.initialFocusedIndex;
+      // reset focusedItemIndex when setKey changes
+      this.setState({
```

Comment:
Can you add a test to cover this so it doesn't break?

Reply:
sure, added a test